category_theory/limits/functor_category.lean

```diff
@@ -0,0 +1,198 @@
+-- Copyright (c) 2018 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+
+import category_theory.products
+import category_theory.limits
+import category_theory.limits.preserves
+
+open category_theory
+
+namespace category_theory.limits
+
+universes u v
+
+variables {C : Type u} [𝒞 : category.{u v} C]
+include 𝒞
+
+variables {J K : Type v} [small_category J] [small_category K]
+
+def switched (F : J ⥤ (K ⥤ C)) : K ⥤ (J ⥤ C) :=
```

Comment:
Ths should be also in `category_theory.functor`. And call it `flip` to be similar to `_root_.flip`.
I hope by being in the namespace one just writes `F.flip`.

Reply:
So it should be protected, then there is no problem with name clashes

Reply:
I've renamed it to `flip`, moved it to `functor_category.lean` (in the `category_theory.functor` namespace), and protected it. It now works great writing `F.flip`!